client/jetpack-cloud/sections/partner-portal/credit-card-fields/index.js

```diff
@@ -0,0 +1,187 @@
+import CreditCardNumberField from './credit-card-number-field';
+import CreditCardExpiryField from './credit-card-expiry-field';
+import CreditCardCvvField from './credit-card-cvv-field';
+import PaymentMethodImage from 'calypso/jetpack-cloud/sections/partner-portal/credit-card-fields/payment-method-image';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export default function CreditCardFields() {
+	const translate = useTranslate();
+	const theme = useTheme();
+	const onEvent = useEvents();
+	// eslint-disable-next-line no-unused-vars
+	const [ isStripeFullyLoaded, setIsStripeFullyLoaded ] = useState( false );
+	const fields = useSelect( ( select ) => select( 'credit-card' ).getFields() );
+	const getField = ( key ) => fields[ key ] || {};
+	const getFieldValue = ( key ) => getField( key ).value ?? '';
+	const getErrorMessagesForField = ( key ) => {
+		const managedValue = getField( key );
+		if ( managedValue?.isRequired && managedValue?.value === '' ) {
+			return [ translate( 'This field is required.' ) ];
+		}
+		return managedValue.errors ?? [];
+	};
+	const { setFieldValue, changeBrand, setCardDataError, setCardDataComplete } = useDispatch(
+		'credit-card'
+	);
+
+	const cardholderName = getField( 'cardholderName' );
+	const cardholderNameErrorMessages = getErrorMessagesForField( 'cardholderName' ) || [];
+	const cardholderNameErrorMessage = cardholderNameErrorMessages?.[ 0 ] || null;
+
+	const cardholderEmail = getField( 'cardholderEmail' );
+	const cardholderEmailErrorMessages = getErrorMessagesForField( 'cardholderEmail' ) || [];
+	const cardholderEmailErrorMessage = cardholderEmailErrorMessages?.[ 0 ] || null;
+
+	const cardholderPhone = getField( 'cardholderPhone' );
+	const cardholderPhoneErrorMessages = getErrorMessagesForField( 'cardholderPhone' ) || [];
+	const cardholderPhoneErrorMessage = cardholderPhoneErrorMessages?.[ 0 ] || null;
+
+	const handleStripeFieldChange = ( input ) => {
+		setCardDataComplete( input.elementType, input.complete );
+		if ( input.elementType === 'cardNumber' ) {
+			changeBrand( input.brand );
+		}
+
+		if ( input.error && input.error.message ) {
+			onEvent( {
+				type: 'a8c_checkout_stripe_field_invalid_error',
+				payload: {
+					type: 'Stripe field error',
+					field: input.elementType,
+					message: input.error.message,
+				},
+			} );
+			setCardDataError( input.elementType, input.error.message );
+			return;
+		}
+		setCardDataError( input.elementType, null );
+	};
+
+	const { formStatus } = useFormStatus();
+	const isDisabled = formStatus !== FormStatus.READY;
+
+	const stripeElementStyle = {
+		base: {
+			fontSize: '14px',
+			color: theme.colors.textColor,
+			fontFamily: theme.fonts.body,
+			fontWeight: theme.weights.normal,
+			'::placeholder': {
+				color: theme.colors.placeHolderTextColor,
+			},
+		},
+		invalid: {
+			color: theme.colors.textColor,
+		},
+	};
+
+	return (
+		<>
+			<CardHeading>{ translate( 'Credit card details' ) }</CardHeading>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Stored Credit Card Details' )` *ES Score: 9*